osde2e/splunk_forwarder_operator_tests.go

```diff
@@ -0,0 +1,145 @@
+		for _, serviceName := range serviceNames {
+			err = k8s.Get(ctx, serviceName, operatorNamespace, &corev1.Service{})
+			Expect(err).Should(BeNil(), "unable to get service %s/%s", operatorNamespace, serviceName)
+		}
+		client, err := openshift.New()
+		Expect(err).NotTo(HaveOccurred(), "openshift client could not be created")
+
+		ginkgo.By("checking the deployment exists and is available")
+		assertions.EventuallyDeployment(ctx, client, deploymentName, operatorNamespace)
+
+		ginkgo.By("checking the operator lock file config map exists")
+		assertions.EventuallyConfigMap(ctx, client, operatorLockFile, operatorNamespace).WithTimeout(time.Duration(300)*time.Second).WithPolling(time.Duration(30)*time.Second).Should(Not(BeNil()), "configmap %s should exist", operatorLockFile)
+
+		ginkgo.By("checking the operator CSV exists")
+		restConfig, _ = config.GetConfig()
+		dynamicClient, err := dynamic.NewForConfig(restConfig)
+		Expect(err).ShouldNot(HaveOccurred(), "failed to configure Dynamic client")
+		EventuallyCsv(ctx, *dynamicClient, operatorName, operatorNamespace).WithTimeout(time.Duration(300)*time.Second).WithPolling(time.Duration(30)*time.Second).Should(BeTrue(), "CSV %s should exist and have Succeeded status", operatorName)
+
+		// TODO: post osde2e-common library add upgrade check
+		//checkUpgrade(helper.New(), "openshift-splunk-forwarder-operator",
+		//	"openshift-splunk-forwarder-operator", "splunk-forwarder-operator",
+		//	"splunk-forwarder-operator-catalog")
+	})
+
+	sf := makeMinimalSplunkforwarder(testsplunkforwarder)
+
+	ginkgo.It("admin should be able to create SplunkForwarders CR", func(ctx context.Context) {
+		err := k8s.WithNamespace(operatorNamespace).Create(ctx, &sf)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("admin should be able to delete SplunkForwarders CR", func(ctx context.Context) {
+		err := k8s.Delete(ctx, &sf)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("dedicated admin should not be able to manage SplunkForwarders CR", func(ctx context.Context) {
+		dsf := makeMinimalSplunkforwarder(dedicatedadminsplunkforwarder)
+		impersonatedResourceClient, _ := k8s.Impersonate("test-user@redhat.com", "dedicated-admins")
+		Expect(sfv1alpha1.AddToScheme(impersonatedResourceClient.GetScheme())).Should(BeNil(), "unable to register sfv1alpha1 api scheme")
+		err := impersonatedResourceClient.WithNamespace(operatorNamespace).Create(ctx, &dsf)
+		Expect(apierrors.IsForbidden(err)).To(BeTrue())
+	})
+
+})
+
+// Create test splunkforwarder CR definition
+func makeMinimalSplunkforwarder(name string) sfv1alpha1.SplunkForwarder {
+	sf := sfv1alpha1.SplunkForwarder{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: operatorNamespace,
+		},
+		Spec: sfv1alpha1.SplunkForwarderSpec{
+			SplunkLicenseAccepted: true,
+			UseHeavyForwarder:     false,
+			SplunkInputs: []sfv1alpha1.SplunkForwarderInputs{
+				{
+					Path: "",
+				},
+			},
+		},
+	}
+	return sf
```

Comment:
Just a suggestion to remove storing the custom resource into a variable and just return the custom resource directly.

```suggestion
	return sfv1alpha1.SplunkForwarder{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: operatorNamespace,
		},
		Spec: sfv1alpha1.SplunkForwarderSpec{
			SplunkLicenseAccepted: true,
			UseHeavyForwarder:     false,
			SplunkInputs: []sfv1alpha1.SplunkForwarderInputs{
				{
					Path: "",
				},
			},
		},
	}
```